src/handlers/gestures/GestureMonitor.tsx

```diff
@@ -1,93 +1,399 @@
+            event.state === State.FAILED ||
+            event.state === State.CANCELLED
+          ) {
+            if (gesture.isOnEndWorklet) {
+              gesture.onEnd?.(event, false);
+            } else if (gesture.onEnd) {
+              console.warn('Animated gesture callback must be a worklet');
+            }
+          }
         } else {
-          RNGestureHandlerModule.attachGestureHandler(
-            gesture.handlerTag,
-            RNRenderer.findHostInstance_DEPRECATED(this)._nativeTag,
-            true
-          );
+          if (gesture.isOnUpdateWorklet) {
+            gesture.onUpdate?.(event);
+          } else if (gesture.onUpdate) {
+            console.warn('Animated gesture callback must be a worklet');
+          }
         }
       }
     }
+  };
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const event = Reanimated.useEvent(
+    callback,
+    ['onGestureHandlerStateChange', 'onGestureHandlerEvent'],
+    true
+  );
+
+  preparedGesture.animatedEventHandler = event;
+  preparedGesture.animatedHandlers = sharedHandlersCallbacks;
+}
+
+interface GestureMonitorProps {
+  gesture?: InteractionBuilder | GestureType;
+  animatedGesture?: InteractionBuilder | GestureType;
+}
+export const GestureMonitor: React.FunctionComponent<GestureMonitorProps> = (
+  props
+) => {
+  const useAnimated = props.animatedGesture != null;
+  const gestureConfig = props.animatedGesture ?? props.gesture;
+  const gesture = gestureConfig?.configure?.() ?? [];
+  const viewRef = useRef(null);
+  const firstRenderRef = useRef(true);
+
+  const preparedGesture = React.useRef<GestureConfigReference>({
+    config: gesture,
+    animatedEventHandler: null,
+    animatedHandlers: null,
+    firstExecution: true,
+    useAnimated: useAnimated,
+  }).current;
+
+  if (useAnimated !== preparedGesture.useAnimated) {
+    throw new Error(
+      'You cannot change whether you are using gesture or animatedGesture while the app is running'
+    );
   }
 
-  render() {
-    if (this.props.gesture?.animatedEventHandler) {
-      return (
-        <AnimatedWrap
-          ref={this.refHandler}
-          onGestureHandlerEvent={this.props.gesture?.animatedEventHandler}>
-          {this.props.children}
-        </AnimatedWrap>
-      );
+  if (preparedGesture.firstExecution) {
+    gestureConfig?.initialize?.();
+  }
+
+  if (useAnimated) {
+    // Whether animatedGesture or gesture is used shouldn't change
+    // during while an app is running
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useAnimatedGesture(preparedGesture);
+  }
+
+  useEffect(() => {
+    firstRenderRef.current = true;
+    const viewTag = findNodeHandle(viewRef.current) as number;
+    attachHandlers(
+      preparedGesture,
+      gestureConfig,
+      gesture,
+      viewTag,
+      useAnimated
+    );
+
+    return () => {
+      dropHandlers(preparedGesture);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!firstRenderRef.current) {
```

Comment:
Are you checking if it's falsy or false here?

Reply:
I'm checking if it's false.